sdk/core/core-rest-pipeline/src/policies/multipartPolicy.ts

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { randomUUID, stringToUint8Array } from "@azure/core-util";
+import { BlobLike, BodyPart, HttpHeaders, PipelineRequest } from "../interfaces";
+import { PipelinePolicy } from "../pipeline";
+import { toStream, concatenateStreams } from "../util/stream";
+import { isInMemoryBlob, isStreamableBlob } from "../util/typeGuards";
+
+function generateBoundary(): string {
+  return `----AzSDKFormBoundary${randomUUID()}`;
+}
+
+function encodeHeaders(headers: HttpHeaders): string {
+  return Array.from(headers)
```

Comment:
From a perf standpoint, wouldn't just iterating over the headers and destructuring into a single string to be faster than the copying into a new array every array.map?

Reply:
Agree that it'd be nice to minimize the number of allocations here since this code is on the hot path.

Reply:
Done!